docs/reference/modules/ROOT/pages/transactions.adoc

```diff
@@ -3,113 +3,74 @@
-the event log. The document itself is submitted to the `doc-topic`,
-using its content hash as key. In Kafka, the `doc-topic` is compacted,
-which enables later deletion of documents.
-
-[#valid-ids]
-== Valid IDs
-
-The following types of `:crux.db/id` are allowed:
-
-* Keyword (e.g. `{:crux.db/id :my-id}` or `{:crux.db/id :dbpedia.resource/Pablo-Picasso}`)
-* String (e.g. `{:crux.db/id "my-id"}`)
-* Integers/Longs (e.g. `{:crux.db/id 42}`)
-* UUID (e.g. `{:crux.db/id #uuid "6f0232d0-f3f9-4020-a75f-17b067f41203"}`)
-* URI (e.g. `{:crux.db/id #crux/id "mailto:crux@juxt.pro"}`)
-* URL (e.g. `{:crux.db/id #crux/id "https://github.com/juxt/crux"}`), including `http`, `https`, `ftp` and `file` protocols
-* Maps (e.g. `{:crux.db/id {:this :id-field}}`) (Note: see https://github.com/juxt/crux/issues/362[issue #362]).
-
-The `#crux/id` reader literal will take URI/URL strings and attempt to coerce them into valid IDs.
-
-URIs and URLs are interpreted using Java classes (java.net.URI and java.net.URL respectively) and therefore you can also use these directly.
-
 [#operations]
 == Operations
+There are four transaction (write) operations:
+
+.Write Operations
+[cols="3,5,^2"]
+|===
+|Operation|Purpose|Pre-condition?
+
+|<<#put,`crux.tx/put`>>|Write a version of a document|
+|<<#delete,`crux.tx/delete`>>|Deletes a specific document|
+|<<#match,`crux.tx/match`>>|Check the document state against the given document|✓
+|<<#evict,`crux.tx/evict`>>|Evicts a document entirely, including all historical versions|
+|<<#transaction-functions,`crux.tx/fn`>>|Runs a transaction function|✓
+|===
 
 [#put]
 === Put
 
-Put's a document into Crux. If a document already exists with the
-given `:crux.db/id`, a new version of this document will be created at
-the supplied `valid time`.
+Put's a document into Crux. If a document already exists with the given `:crux.db/id`, a new version of this document will be created at the supplied `valid time`.
 
 [source,clojure]
 ----
 [:crux.tx/put
  {:crux.db/id :dbpedia.resource/Pablo-Picasso :first-name :Pablo} <1>
- #inst "2018-05-18T09:20:27.966-00:00"] <2>
+ #inst "2018-05-18T09:20:27.966-00:00" <2>
+ #inst "2018-05-19T08:31:15.966-00:00"] <3>
 ----
 
-<1> The document itself. Note that the ID must be included as part of the
-document.
-<2> `valid time`
-
-Note that `valid time` is optional and defaults to transaction time,
-which is taken from the Kafka log.
-
-Crux `puts` into the past at a single point by default, so to overwrite several
-versions across a range in valid time, you can either submit a transaction
-containing several operations or supply a third argument to specify an `end
-valid time`. This period is inclusive-exclusive, such that the start of the
-validity period is included in the validity range, while the end is excluded.
+<1> Document to add
+<2> (optional) <<#valid-time,`valid time`>>
+<3> (optional) <<#end-valid-time,`end valid time`>>
 
```

Comment:
Maybe a note here to refer to the valid time section L197 for operational effect?

Reply:
Ditto my above